fred2/volumetricsdlg.cpp

```diff
@@ -0,0 +1,391 @@
+	static constexpr char* Tooltip_steps = _T("If you see banding on ships in the volumetrics, increase this.");
+	static constexpr char* Tooltip_oversampling = _T("Increasing this improves the nebula's edge's smoothness especially for large nebula at low resolutions.");
+	static constexpr char* Tooltip_henyey = _T("Values greater than 0 cause a cloud-like light shine-through, values smaller than 0 cause a highly reflective nebula.");
+	static constexpr char* Tooltip_sun_falloff = _T("Values greater than 1 means the nebula's depths are brighter than they ought to be, values smaller than 0 means they're darker.");
+	static constexpr char* Tooltip_steps_sun = _T("If you see banding in the volumetrics' light and shadow, increase this.");
+	static constexpr char* Tooltip_em_spread = _T("How much light sources will scatter in the nebula.");
+	static constexpr char* Tooltip_em_falloff = _T("Values greater than 1 will cause light sources in the nebula to spread more evenly, values smaller than 1 will cause them to be more intense in their center.");
+	m_toolTip.AddTool(GetDlgItem(IDC_OPACITY_DISTANCE), Tooltip_distance);
+	m_toolTip.AddTool(GetDlgItem(IDC_SPIN_OPACITY_DISTANCE), Tooltip_distance);
+	m_toolTip.AddTool(GetDlgItem(IDC_STEPS), Tooltip_steps);
+	m_toolTip.AddTool(GetDlgItem(IDC_SPIN_STEPS), Tooltip_steps);
+	m_toolTip.AddTool(GetDlgItem(IDC_OVERSAMPLING), Tooltip_oversampling);
+	m_toolTip.AddTool(GetDlgItem(IDC_SPIN_OVERSAMPLING), Tooltip_oversampling);
+	m_toolTip.AddTool(GetDlgItem(IDC_HGCOEFF), Tooltip_henyey);
+	m_toolTip.AddTool(GetDlgItem(IDC_SPIN_HGCOEFF), Tooltip_henyey);
+	m_toolTip.AddTool(GetDlgItem(IDC_SUN_FALLOFF), Tooltip_sun_falloff);
+	m_toolTip.AddTool(GetDlgItem(IDC_SPIN_SUN_FALLOFF), Tooltip_sun_falloff);
+	m_toolTip.AddTool(GetDlgItem(IDC_STEPS_SUN), Tooltip_steps_sun);
+	m_toolTip.AddTool(GetDlgItem(IDC_SPIN_STEPS_SUN), Tooltip_steps_sun);
+	m_toolTip.AddTool(GetDlgItem(IDC_EM_SPREAD), Tooltip_em_spread);
+	m_toolTip.AddTool(GetDlgItem(IDC_SPIN_EM_SPREAD), Tooltip_em_spread);
+	m_toolTip.AddTool(GetDlgItem(IDC_EM_FALLOFF), Tooltip_em_falloff);
+	m_toolTip.AddTool(GetDlgItem(IDC_SPIN_EM_FALLOFF), Tooltip_em_falloff);
+	m_toolTip.Activate(TRUE);
+
+	if (The_mission.volumetrics) {
+		const volumetric_nebula& volumetrics = *The_mission.volumetrics;
+		m_enabled = true;
+		m_volumetrics_hull = volumetrics.hullPof.c_str();
+		m_position = volumetrics.pos;
+		m_color = { static_cast<int>(std::get<0>(volumetrics.nebulaColor) * 255.0f),
+					static_cast<int>(std::get<1>(volumetrics.nebulaColor) * 255.0f),
+					static_cast<int>(std::get<2>(volumetrics.nebulaColor) * 255.0f) };
+		m_opacity = volumetrics.alphaLim;
+		m_distance = volumetrics.visibility;
+		m_steps = volumetrics.steps;
+		m_resolution = volumetrics.resolution;
+		m_oversampling = volumetrics.oversampling;
+		m_henyeyGreenstein = volumetrics.henyeyGreensteinCoeff;
+		m_sunFalloffFactor = volumetrics.globalLightDistanceFactor;
```

Comment:
Same question about these two

Reply:
This one, just like ``m_opacity = volumetrics.alphaLim;`` has a bit of an issue with "userfacing name" vs "technical name".
``globalLightDistanceFactor`` is a reasonably accurate name for what the variable actually affects within the shaders, namely serving as a factor for the distance travelled by the global light in the nebula.
In practice, while descriptive to a developer, this is very unintuitive to an end user. For an end user "Sun Falloff" seems easier to understand (I hope), because it describes what happens visually.
So I am not quite sure.
It would certainly be possible to rename the variable here, but then we have a variable that doesn't quite fit the name of the label next to its text input, and I am unsure about what I prefer there.

Reply:
I'm fine with keeping it as-is